crates/parser/src/ast/lex.rs

```diff
@@ -29,6 +29,7 @@ pub struct Span {
 pub enum Token {
     Whitespace,
     Comment,
+    DocComment,
```

Comment:
Personally at least I'd prefer to keep the lexer the same as before where doc-comments are simply lex'd as normal comments and it's up to other systems externally to interpret doc-comments and process them.

Reply:
Okay! 

My first idea would be to change then `Resolver::docs` only, and no changes to lex. But then, I am not sure what should be returned by [parser::Resolver::docs](https://github.com/bytecodealliance/wit-bindgen/pull/125/files#diff-2164780efe09de34f76096a0bcfdefb5b1f4226ce229361f647c9174335a7961L511-R511), as it would return only doc-comments and there would be no access point to normal comments.

> it's up to other systems externally to interpret doc-comments and process them.

Is the `parser::Resolver` considered an "external system"? 

The `parser` crate only defines the `Doc` struct. To make comments available to external crates, should I define a `Comments` struct and define a new method `Resolver::comments`?

Reply:
Yeah the original intention was to discard normal comments and only pass through doc-comments, I unfortunately just never got around to it.

Reply:
Also, yes, by "external system" I basically mean external to this module, anything in `Resolver` is fine.